File: VB/WindowsApplication12/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
' ...

Namespace WindowsApplication12
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Public currentPage As Integer

		Private Sub xrLabel2_PrintOnPage(ByVal sender As Object, ByVal e As PrintOnPageEventArgs) Handles xrLabel2.PrintOnPage
			currentPage = e.PageIndex
		End Sub

	End Class
End Namespace

File: CS/WindowsApplication12/XtraReport1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
// ...

namespace WindowsApplication12 {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        public int currentPage;

        private void xrLabel2_PrintOnPage(object sender, PrintOnPageEventArgs e) {
            currentPage = e.PageIndex;
        }

    }
}